src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -679,19 +682,22 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return undefined;
 	}
 
-	public mergeCells(cell: ICellModel, secondCell: ICellModel, newLinesRemoved: string[] | undefined): void {
-		let index = this._cells.findIndex(cell => cell.equals(cell));
-		if (index > -1) {
-			cell.source = newLinesRemoved.length > 0 ? [...cell.source, ...newLinesRemoved, ...secondCell.source] : [...cell.source, ...secondCell.source];
-			cell.isEditMode = true;
-			// Set newly created cell as active cell
-			this.updateActiveCell(cell);
-			this._contentChangedEmitter.fire({
-				changeType: NotebookChangeType.CellsModified,
-				cells: [cell],
-				cellIndex: index
-			});
-			this.deleteCell(secondCell, false);
+	public mergeCells(cells: ICellModel[], newLinesRemoved: string[] | undefined): void {
```

Comment:
I don't think newLinesRemoved can actually be undefined here so remove the `| undefined`. (and if it was possible to be undefined then your logic below would throw in that case so be careful)